### ISSUE TITLE
Slow parsing on complex workbook

### ISSUE THREAD

**Issue by boosth** [vinci1it2000/formulas#75]
I am experiencing slow parsing on a complex workbook. The workbook loads in a few seconds on a desktop app but it takes 12 minutes for the parser to complete the initial load/compile process.

Specifically this function from excel/__init__.py in add_cell()

cell = Cell(crd, val, context=ctx, check_formula=check_formula).compile(
            references=references, context=ctx
        )

I understand from another issue where you added from_range() function that it is related to the number of cells in each sheet but even still it seems to take an excess amount of time to load/compile when compared to a desktop app or even a javascript web UI.

I have run the same code on multiple hardware specs and they all have approximately the same load speed which suggests the process is relying on a single processor/thread.

Is there a recommended method for quicker loading or are there any logic improvements that could be added to the code?

**Comment by boosth** [comment]
The schedulas library which is a dependency of Formulas library has support for parallel processing. However, it is not enabled by default in the formulas library or mentioned in the formulas docs.

https://github.com/vinci1it2000/schedula#asynchronous-and-parallel-dispatching

to enable it you will need to install this additional library.

pip install multiprocess

Schedulas provides four default executors to dispatch asynchronously or in parallel:

async: execute all functions asynchronously in the same process,
parallel: execute all functions in parallel excluding :class:`~schedula.utils.dsp.SubDispatch` functions,
parallel-pool: execute all functions in parallel using a process pool excluding :class:`~schedula.utils.dsp.SubDispatch` functions,
parallel-dispatch: execute all functions in parallel including :class:`~schedula.utils.dsp.SubDispatch`.


ex.  xl_model.calculate(executor="async")
 

As the docs say there is a cost to running in parallel so YMMV.  Depending on the size of the input file it may exceed available memory and also may incur additional processing time.   

For best results it is better to use a minimal input file instead of large complex input files.

**Comment by vinci1it2000** [comment]
The last release has improved the parsing of excel sheets [skipping](https://github.com/vinci1it2000/formulas/commit/8d95a4e5fcda62ea0882da5da1945b44cff8b9de#diff-8fc2f6af9119782e131fca3d7f7e01eb000ebeb6ffb0b08b31fb39b27b114374R283-R297) some cells already parsed.
